docs/source/configuration/traffic-shaping.mdx

```diff
@@ -44,7 +33,135 @@ traffic_shaping:
+per subgraph, not per backend host.
+
+```yaml title="router.yaml"
+traffic_shaping:
+  all:
+    global_rate_limit: # Accept a maximum of 10 requests per 5 secs. Excess requests must be rejected.
+      capacity: 10
+      interval: 5s # Must not be greater than 18_446_744_073_709_551_615 milliseconds and not less than 0 milliseconds
+```
+
+### Experimental request retry
 
+On failure, subgraph requests can be retried automatically. This is deactivated by default for mutations.
+This uses [Finagle's *RetryBudget* algorithm](https://finagle.github.io/blog/2016/02/08/retry-budgets/),
+in which every successful request adds an expirable token to a bucket, and every retry consumes a number
+of those tokens. On top of that, a minimal number of retries per second is available, to test regularly
+when the retry budget was entirely consumed or on startup when very few requests have been sent.
+The tokens expire so the budget has a large number of available retries if a lot of recent requests were
+successful, but reduces quickly on frequent failures to avoid sending too much traffic to the subgraph.
+
+It is configurable as follows:
+
+```yaml title="router.yaml"
+traffic_shaping:
+  all:
+    experimental_retry:
+      min_per_sec: 10 # minimal number of retries per second (`min_per_sec`, default is 10 retries per second)
+      ttl: 10s # for each successful request, we register a token, that expires according to this option (default: 10s)
+      retry_percent: 0.2 # defines the proportion of available retries to the current number of tokens
+      retry_mutations: false # allows retries on mutations. This should only be enabled if mutations are idempotent
 ```
 
-Any configuration under the `subgraphs` key takes precedence over configuration under the `all` key. In the example above, query deduplication is enabled for all subgraphs _except_ the `products` subgraph.
+### Variable deduplication
+
+In federated queries, using the `_entities` operation, it is often the case that one entity is requested from
+multiple parts of a query. As an example, for each review of each product, geting the author's name might return
+the same name multiple times if an author wrote multiple reviews.
+
+To reduce the size of subgraph requests and the amount of work they might perform, the list of entities sent
+can be deduplicated. This is activated with the `deduplicate_variables` option:
+
+```yaml title="router.yaml"
+traffic_shaping:
+  deduplicate_variables: true # Enable the variable deduplication optimization.
+```
+
+### Query deduplication
+
+If similar queries are sent to a subgraph, the Apollo Router can avoid sending them multiple times, and instead
+copy the result of the first query when it comes back from the subgraph, thus reducing the traffic to the subgraph
+and the overall client request latency.
+
+The subgraph queries have to be exactly the same (path, headers and body) to be deduplicated.
+
```

Comment:
```suggestion
If the Router is simultaneously processing similar queries, it may result in producing multiple identical requests to a subgraph.  With the `deduplicate_query` functionality enabled, the Router can avoid sending the same query multiple times and instead buffer one or more of the dependent queries pending the result of the first, and reuse that result to fulfill all of the initial queries.  This will reduce the overall traffic to the subgraph and the overall client request latency.  To meet the criteria for deduplication, the feature must be enabled and the subgraph queries must have have the same HTTP path, headers and body:

```